src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(FIND_BY_ID)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(takeBook(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(FIND_ALL)) {
+            List<Book> books = new ArrayList<>();
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(takeBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find books!", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(UPDATE)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Can't update book. Book with id "
+                        + book.getId() + " was not found!");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(DELETE_BY_ID)) {
+            statement.setLong(1, id);
+            int deletedRows = statement.executeUpdate();
+            if (deletedRows == 0) {
+                throw new DataProcessingException("No book found with id: " + id);
+            }
+            return deletedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book with id: " + id, e);
+        }
+    }
+
+    private Book takeBook(ResultSet resultSet) throws SQLException {
+        Long bookId = resultSet.getLong("id");
+        String title = resultSet.getString("title");
+        BigDecimal price = resultSet.getBigDecimal("price");
+        Book book = new Book();
+        book.setId(bookId);
+        book.setTitle(title);
+        book.setPrice(price);
+        return book;
+    }
```

Comment:
To avoid code duplication when working with `ResultSet`, move the logic of retrieving data from the `ResultSet` into the `Book` entity to a separate private method, which you have done correctly here. Ensure you use this method in all places where you convert `ResultSet` to `Book`.